client/keys.go

```diff
@@ -468,6 +470,21 @@ func (k *Key) CertDERBytes() []byte {
 	return k.cert.Raw
 }
 
+// SetCert assigns the provided certificate to the key after verifying it matches the key.
+func (k *Key) SetCert(cert *x509.Certificate) error {
+	certPubKey, ok := cert.PublicKey.(crypto.PublicKey)
+	if !ok {
+		return errors.New("certificate public key is not the appropriate type")
+	}
+
+	if !reflect.DeepEqual(certPubKey, k.pubKey) {
```

Comment:
I think we already have a method to compare two `crypto.PublicKey`s

https://github.com/google/go-tpm-tools/blob/66ec981c39007db9718f14f34c15102cba642499/server/verify.go#L125

We should move that method from `server` to `internal` and then just use it here